### ISSUE TITLE
Scopes don't work with subclasses

### ISSUE THREAD

**Issue by jafuentest** [douglaslise/mongoid_enumerable#1]
Steps to reproduce:

Define the following classes:

```
class Transport
  include Mongoid::Document
  include MongoidEnumerable

  enumerable :terrain, %w[land water]
end

class Car < Transport; end

class Boat < Transport; end
```

Now, running:

`Car.all` gives the following criteria
```
 => #<Mongoid::Criteria
  selector: {}
  options:  {}
  class:    Car
  embedded: false>
```

`Car.land` gives the following criteria
```
 => #<Mongoid::Criteria
  selector: {"terrain"=>"land"}
  options:  {}
  class:    Transport
  embedded: false>
```

Notice how `Car.land` sets the selector `{"terrain"=>"land"}` correctly, but sets the class to the superclass Transport, so `Car.land` will give you all Vehicles of land terrain instead of just cars with land terrain

I'll try to see if I can fix it myself and send you a PR

**Comment by douglaslise** [comment]
Thank you very much @jafuentest by the fix. Be welcome to contribute here.

**Comment by douglaslise** [comment]
Closed by pull request #2
